From c1e1f5dcd936b5e867cfc9f5edf88a1e5aa6940f Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 13 Aug 2021 12:57:48 +0300
Subject: [PATCH] [SW-0] puts: fix the error strings output

Some error prints were overwritten by next print out due to those not
having the '\n' character at the end of the string.
Add the new line character to fix.

Change-Id: I0b8727905bacab799bf923484807bacef905c09c
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/common.c b/arch/gaudi/common.c
index cbaad5b..ff7f54e 100644
--- a/arch/gaudi/common.c
+++ b/arch/gaudi/common.c
@@ -237,7 +237,7 @@
 		 * timeout is used
 		 */
 		if (timeout_expired(start, PCIE_EXT_LD_TIMEOUT_US)) {
-			preboot_puts("Timed-out waiting for ext_ld_done!");
+			preboot_puts("Timed-out waiting for ext_ld_done!\n");
 			break;
 		}
 	}
diff --git a/arch/gaudi/hbm.c b/arch/gaudi/hbm.c
index c6fb2f5..0982e88 100644
--- a/arch/gaudi/hbm.c
+++ b/arch/gaudi/hbm.c
@@ -1395,7 +1395,7 @@
 		else
 			ret = dma_ch;
 	} else {
-		preboot_puts("hbm_pc: Illegal input");
+		preboot_puts("hbm_pc: Illegal input\n");
 		return -1;
 	}
 
@@ -1465,7 +1465,7 @@
 	int dma_ch;
 	struct hbm *hbm = &hbm_data;
 
-	preboot_puts("perform row repair");
+	preboot_puts("perform row repair\n");
 
 	if (hbm->row_replacement_iter[device] == MAX_ROW_REPLACEMENT_ITERATIONS) {
 		preboot_puts("Max row replacement iteration reached\n");
diff --git a/arch/greco/common.c b/arch/greco/common.c
index 36d05ba..0a4724d 100644
--- a/arch/greco/common.c
+++ b/arch/greco/common.c
@@ -299,7 +299,7 @@
 		 * timeout is used
 		 */
 		if (timeout_expired(start, PCIE_EXT_LD_TIMEOUT_US)) {
-			preboot_puts("Timed-out waiting for ext_ld_done!");
+			preboot_puts("Timed-out waiting for ext_ld_done!\n");
 			break;
 		}
 	}
diff --git a/preboot/fit.c b/preboot/fit.c
index 78cac6c..0eafde8 100644
--- a/preboot/fit.c
+++ b/preboot/fit.c
@@ -1038,7 +1038,7 @@
 	if (conf >= 0) {
 		cell = fdt_getprop(fit, conf, FIT_LOAD_TYPE_PROP, &len);
 		if (!cell) {
-			preboot_puts("load type does not exist!");
+			preboot_puts("load type does not exist!\n");
 			*load_type = type;
 			return 0;
 		}
@@ -1080,13 +1080,13 @@
 
 	images_noffset = fdt_path_offset(fit, FIT_IMAGES_PATH);
 	if (images_noffset < 0) {
-		preboot_puts("Can't find images parent node");
+		preboot_puts("Can't find images parent node\n");
 		return images_noffset;
 	}
 
 	noffset = fdt_subnode_offset(fit, images_noffset, image_uname);
 	if (noffset < 0)
-		preboot_puts("Can't get node offset for image unit name");
+		preboot_puts("Can't get node offset for image unit name\n");
 
 	return noffset;
 }
